File: node/zone_records.js
const { DNSimple, AuthenticationError } = require('dnsimple');

(async () => {
  const dnsimple = new DNSimple({
    baseUrl: 'https://api.sandbox.dnsimple.com',
    accessToken: process.env.TOKEN,
    userAgent: 'dnsimple-examples',
  });

  try {
    const identity = await dnsimple.identity.whoami();
    const accountId = identity.data.account.id;
    const zoneName = process.env.DOMAIN || 'dns.com';

    // List all of the records for a zone
    const zoneRecords = await dnsimple.zones.listZoneRecords.collectAll(accountId, zoneName);
    console.log(zoneRecords);
  } catch (err) {
    if (err instanceof AuthenticationError) {
      console.error('Authentication error. Check your token is correct for the sandbox environment.');
    } else {
      console.error(err);
    }
  }
})();

File: elixir/zone_records.exs
# Usage: mix run zone_record.exs example.com

Dnsimple.start

client = %Dnsimple.Client{
  access_token: Application.get_env(:dnsimple, :access_token),
  base_url: "https://api.sandbox.dnsimple.com/"
}

# Grabs the first account based on your user token
account = case Dnsimple.Identity.whoami(client) do
  {:ok, response} -> response.data.account
  {:error, error} -> raise RuntimeError, message: error.message
end

name = List.last(System.argv)

# List all the DNS records for the given domain
case Dnsimple.Zones.list_zone_records(client, account["id"], name) do
  {:ok, response} -> IO.inspect response.data
  {:error, error} -> raise RuntimeError, message: error.message
end
